### ISSUE TITLE
Which version of OpenAPI spec does `Validate(context.Context, ...ValidationOption)` method validate against?

### ISSUE THREAD

**Issue by andrewrowe-wk** [getkin/kin-openapi#812]
Hello!

When using the `Validate(context.Context, ...ValidationOption)` method to validate a loaded openAPI spec, which version of the spec is it validating against? From what I can tell the validation rules are hard-coded in the logic of the Go code, as opposed to validating against a particular version of the schema (for example: https://spec.openapis.org/oas/3.0/schema/2021-09-28). Am I correct in this?

**Comment by fenollp** [comment]
Yes but there is ongoing work to be able to validate openapiv3 documents against other versions of the openapiv3 spec => https://github.com/getkin/kin-openapi/pull/744
